src/monattack.cpp

```diff
@@ -2875,6 +2875,11 @@ bool mattack::grab( monster *z )
     z->add_effect( effect_grabbing, 2_turns );
     target->add_effect( effect_grabbed, 2_turns, body_part_torso, false,
                         prev_effect + z->get_grab_strength() );
+    // if too many entities grab a player they can suffocate
+    // if this is the first monster to grab you set the players oxygen levels
+    if( ( target->is_npc() || target->is_avatar() ) && prev_effect == 0 ) {
+        target->as_character()->oxygen = 30 + 2 * target->as_character()->str_cur;
```

Comment:
```suggestion
        target->as_character()->oxygen = std::min( target->as_character()->oxygen,
                                         30 + 2 * target->as_character()->str_cur );
```
otherwise you can let a monster grab you underwater to replenish your oxygen. And you should probably extract this to a helper so you can deduplicate below in `grab_drag()`

Reply:
Oxygen by default is set to zero so this doesn't work. I'll move it to a character helper function for set_oxygen() that checks if you are already drowning or grabbed before setting the value and put it in the relevant spots.